### DISCUSSION TITLE
font size in output of an R code block

### DISCUSSION THREAD

**Discussion by christophhofer** [quarto-dev/quarto-cli#2358] [Q&A] [answered]
With which code option can the font size be changed (R Code Block). In R Markdown there is the argument size for it. Is there something comparable in Quarto.

**Comment by mcanouil** [comment, marked as answer]
With the `attr-*` or `class-*` YAML options (<https://quarto.org/docs/reference/cells/cells-knitr.html>)

````
---
format: html
---

```{r}
#| attr-source: "style='font-size: 5em'"
#| attr-output: "style='font-size: 0.2em'"
1 + 1
```
````

<img width="1624" alt="screenshot of VSCode showing the rendering of the Quarto document described above this picture" src="https://user-images.githubusercontent.com/8896044/189491731-9e2a7fe6-1abd-43e9-b212-fdd43f69020f.png">

**Reply by christophhofer** [reply]
Unfortunately, this does not work.

**Reply by mcanouil** [reply]
What do you mean "this does not work"? I just showed it working in my example.

**Reply by christophhofer** [reply]
Sorry, I did not make myself clear. The output format is pdf (beamer slides) and not html. For html your solution works.

**Reply by mcanouil** [reply]
Yes, you should have started by properly describing the issue and proposed a reproducible example.

Also, all working `knitr` options should be available.
I just tried with `rmarkdown`/`knitr`, the option you said was working does not with latest version.

````
---
output: pdf_document
---

```{r, size = "footnotesize"}
1 + 1
```
````

Also from <https://yihui.org/knitr/options/#code-decoration>, I can see that the only format mentioned is Sweave "rnw" files.

As a workaround you could use raw LaTeX in your document.

**Reply by christophhofer** [reply]
Yes, that is how I do it. Anyway, thanks for the help.

**Reply by mcanouil** [reply]
You mean you use Sweave?
Or raw LaTeX?

The size option did not work for `rmarkdown`.

To note, depending on what your output is, you could print raw LaTeX in the code cell.
Also, you could make a feature request for a source/output size.

**Reply by christophhofer** [reply]
I use R Markdown (Rmd file) and partly raw LaTex in it.  If I control the font size via raw LaTex (e.g. \tiny) this also has an influence on the R code block.

I just didn't realize that the size parameter in the R chunk doesn't work. I have  I have read over the ending Rnw.

**Reply by mcanouil** [reply]
Unless I am missing something (_i.e., I am not a heavy user of PDF formats with `rmarkdown' or Quarto), I think the solution would be a Lua filter to tweak only the output of a code cell.
I'll try later some ideas.

Could you unmark the answer, since for now, there is no proper answer for PDF?

**Reply by mcanouil** [reply]
@cderv I am pinging you since you surely have more experienced with `knitr`/`rmarkdown` engine to customise code chunk/cell output.
I did not find anything in Quarto's documentation (<https://quarto.org/docs/reference/formats/pdf.html>).
Are you aware of some kind of equivalent to `attr-*` or `class-*` options but for LaTeX/PDF to, in this case, modify the font size?

**Reply by cderv** [reply]
> I just didn't realize that the size parameter in the R chunk doesn't work. I have I have read over the ending Rnw

Yes this is an option that will work for Rnw not Rmd document. There should not be mention of it in R Markdown documentation. 
it is used in **knitr** with specific hooks using for LaTeX generation using Rnw or Rtex document I believe.

> With which code option can the font size be changed (R Code Block). In R Markdown there is the argument size for it. Is there something comparable in Quarto.

Regarding the main question, I don't think we have a configuration to change font size for code that is common to all formats. 

With all HTML output, theming system could be use to change the font of code block I believe using CSS. 
And a custom solution for LaTeX would need to be found. 
Pandoc (and this quarto) has font options for LaTeX that is suppose to help tweak such things (https://quarto.org/docs/reference/formats/pdf.html#fonts) . Otherwise, I think we use something like `\usepackage[scale=.8]{sourcecodepro}` in our preamble of some books, so that the space used for code chunk (monofont) is smaller. 

If we want something on per chunk only, I don't think we have that, and if Pandoc is allowing such tweaks (it is creating a specific Verbatim environment that I don't know if it has options). 

Hope it helps.
